docs/clients/javascript/core.md

```diff
@@ -0,0 +1,107 @@
+
+**Swap for JS HTTP Client at any time**: The JS Core Client and the [JS HTTP Client](./http.md) implement the same [CeramicApi](https://developers.ceramic.network/reference/typescript/interfaces/_ceramicnetwork_common.ceramicapi-1.html){:target="_blank"} TypeScript interface, so swapping between clients is seamless and doesn't require changing your application logic; it only requires changing your setup.
+
+## **Installation**
+Installing the JS Core Client requires a console, [Node.js](https://nodejs.org/en/){:target="_blank"} v14, and [npm](https://www.npmjs.com/get-npm){:target="_blank"} v6. Make sure to have these installed on your machine.
+
+    !!! warning ""
+    While npm v7 is not officially supported, you may be able to get it to work anyway, however you will need to have the `node-pre-gyp` package installed globally:
+    
+    ```bash
+    npm install -g node-pre-gyp
+    ```
+    
+    *This is required until node-webrtc (which IPFS depends on) [is upgraded](https://github.com/node-webrtc/node-webrtc/pull/694).*
+
+### 1. Install the Core client
+Open your console and install the JS Core Client using npm.
+
+``` bash
+npm install @ceramicnetwork/core
+```
+
+### 2. Import the Core client
+
+``` javascript
+import Ceramic from '@ceramicnetwork/core'
+```
+
+### 3. Import IPFS with dag-jose
+Ceramic utilizes the [dag-jose](../../learn/glossary.md#dagjose) IPLD codec to format and store data in IPFS.
+
+``` javascript
+import IPFS from 'ipfs'
+import dagJose from 'dag-jose'
+import { sha256 } from 'multiformats/hashes/sha2'
+import legacy from 'multiformats/legacy'
+
+const hasher = {}
+hasher[sha256.code] = sha256
+const dagJoseFormat = legacy(dagJose, {hashes: hasher})
+
+```
+
+### 4. Create an IPFS instance
+Create an instance of `js-ipfs` with `dag-jose` enabled.
+
+``` javascript
+const ipfs = await Ipfs.create({ ipld: { formats: [dagJoseFormat] } })
+```
+
+### 5. Create a Ceramic instance
+Create an instance of Ceramic by passing ipfs and an optional configuration object.
+
+``` javascript
+const ceramic = await Ceramic.create(ipfs, config)
+```
+
+### 6. Import DID resolvers
+Import resolvers for all DID methods that this Core Client will support. This should be inclusive of the DID Method that you will use for [authentication](../../build/authentication.md), but should also include all other DID Methods for which your node could possibly need to verify signatures. Therefore, it is recommended that all Core Clients be able to resolve at least the `did:3` and `did:key` DID methods.
+
+
+``` javascript
+import KeyDidResolver from 'key-did-resolver'
+import ThreeIdResolver from '@ceramicnetwork/3id-did-resolver'
+```
+
+### 7. Create a resolver instance
```

Comment:
This could be merged with step 8

Reply:
@oed I'm not sure how that would work? Mind proposing a change for what you have in mind?

Reply:
Just make the content here a part of the *Create a DID instance* section below.

Reply:
@oed I made this change, but also in the description of that section we say to optionally include a did provider for use with authentication... can we include that in the example code because it's unclear to me where that would go. Mind adding that?

Reply:
I think that might be covered elsewhere? @stbrody do you know?

Reply:
All of that is covered in the authentication page: https://developers.ceramic.network/build/authentication/

Reply:
sounds good